api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1736,6 +1736,14 @@ type AzurePlatformSpec struct {
 	// +immutable
 	ResourceGroupName string `json:"resourceGroup"`
 
+	// ClientResourceGroupName is the name of an existing resource group provided by a client; it is expected to contain
+	// an existing, pre-setup VNET and Subnet. This field is only used by ARO HCP.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="ClientResourceGroupName is immutable"
```

Comment:
Do these group names have validation applied at the Azure level, and, if so, can we also add them to the CEL here?

Reply:
For `ClientResourceGroupName`, I don't think it matters as much because it needs to pre-exist meaning they've already created it and its passed whatever requirements Azure has.

Reply:
I'll look into this for `ResourceGroupName` though.